### PR TITLE
[Fix](Variant) fix variant serialize root node

### PR DESCRIPTION
**Pull request by eldenmoon**
fix serialize variant root

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by shuke987** [comment]
run performance

File: be/src/vec/common/schema_util.cpp [hunk 1]
```diff
@@ -506,7 +507,7 @@ Status parse_variant_columns(Block& block, const std::vector<int>& variant_pos,
                 auto raw_column = vectorized::ColumnString::create();
                 for (size_t i = 0; i < var->rows(); ++i) {
                     std::string raw_str;
-                    var->serialize_one_row_to_string(i, &raw_str);
+                    static_cast<void>(var->serialize_one_row_to_string(i, &raw_str));
```
Comment (xiaokang) [inline be/src/vec/common/schema_util.cpp:510]: Why not return if error

File: be/src/vec/columns/column_object.cpp [hunk 2]
```diff
@@ -1106,15 +1108,15 @@ bool ColumnObject::serialize_one_row_to_json_format(int row, rapidjson::StringBu
     } else {
         output->Clear();
         rapidjson::Writer<rapidjson::StringBuffer> writer(*output);
-        return root.Accept(writer);
+        CHECK(root.Accept(writer));
```
Comment (xiaokang) [inline be/src/vec/columns/column_object.cpp:1111]: Why not return error status?

File: be/src/vec/common/schema_util.cpp (hunk 1 above)
Comment (eldenmoon) [reply be/src/vec/common/schema_util.cpp]: done

File: be/src/vec/columns/column_object.cpp (hunk 2 above)
Comment (eldenmoon) [reply be/src/vec/columns/column_object.cpp]: done

File: be/src/vec/columns/column_object.cpp [hunk 3]
```diff
@@ -1031,53 +1033,49 @@ void get_json_by_column_tree(rapidjson::Value& root, rapidjson::Document::Alloca
...
+                                                      bool* is_null) const {
     CHECK(is_finalized());
     if (subcolumns.empty()) {
         if (is_null != nullptr) {
             *is_null = true;
         } else {
             rapidjson::Value root(rapidjson::kNullType);
             rapidjson::Writer<rapidjson::StringBuffer> writer(*output);
-            return root.Accept(writer);
+            CHECK(root.Accept(writer));
```
[the remark this answers is not in the document]
Comment (eldenmoon) [reply be/src/vec/columns/column_object.cpp]: done

**Comment by eldenmoon** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

**Pull request merged**